strategoxt/stratego-libraries/java-backend/java/runtime/org/strategoxt/HybridInterpreter.java

```diff
@@ -385,6 +388,16 @@ public void uninit() {
 		SSLLibrary lib = SSLLibrary.instance(getContext());
 		if (lib != null)
 			lib.getIOAgent().closeAllFiles();
+		// Copy and clear before closing, so that classLoaders is empty when an exception is thrown.
+		final ArrayList<URLClassLoader> classLoaders = new ArrayList<>(this.classLoaders);
+		this.classLoaders.clear();
+		for(URLClassLoader classLoader : classLoaders) {
+			try {
+				classLoader.close();
+			} catch(IOException e) {
+				throw new RuntimeException(e);
+			}
+		}
```

Comment:
I think you can avoid the copy by pulling the `try` out around the `for` loop and `clear`ing in a `finally` clause.

Reply:
That is cleaner indeed, fixed.

Reply:
I am wondering now... Perhaps you want to try to close each one, and only afterwards throw any caught exceptions?

Reply:
Thought about that, too much effort for now.

Reply:
Ok, then I'll consider it your problem if that ends up being a bug 🤷‍♂